cli/internal/check/diff.go

```diff
@@ -146,18 +154,29 @@ func formatCheckPrompt(file code.FilteredFile, documents []*docs.FilteredDoc, pr
 	// Add type-of-change specific information
 	switch file.Action {
 	case code.ActionInsert:
-		// Add <file>
-		prompt.WriteString("<file>\n")
-		prompt.WriteString(fmt.Sprintf("  <file_name>%s</file_name>\n", file.Filename))
-		prompt.WriteString("  <file_content>\n")
-		prompt.WriteString(string(file.Contents))
-		prompt.WriteString("\n")
-		prompt.WriteString("  </file_content>\n")
-		prompt.WriteString("</file>\n")
-		prompt.WriteString("\n")
-		// Add prompt
-		prompt.WriteString(fmt.Sprintf("Given that the file %s was created, ", file.Filename))
-		prompt.WriteString("and that the contents of the created file are in <file>, ")
+		if file.Diff != "" {
```

Comment:
We may want to flip this logic here and below so that we use the file contents if we have them, and then use the diff if we don't. My reasoning is that we will use fewer tokens by posting the contents of the file than the diff, as well as the diff in this case feels unnecessary as long as we have the contents.